### ISSUE TITLE
Table toolbar needed

### ISSUE THREAD

**Issue by emilianosantucci** [RoXuS/paper-datatable-api#11]
Hi @RoXuS.
thanks to follow David-Mulder work with your project.

As user of "previous" version, one of useful feature has been [toolbars](http://david-mulder.github.io/paper-datatable/components/paper-datatable/docs/docs.html?getting-started-with-data-table-cards#toolbars).

Do you intend to implement that on your project?

**Comment by RoXuS** [comment]
Hi @emilianosantucci,

Thank you for your interest.

Hum, for me, the `paper-datatable-api` must be an independent component, maybe it is an error ?

As you can see in the [demo](https://roxus.github.io/paper-datatable-api) I have a toolbar to hide/show column, I have added paper-input in [this commit](https://github.com/RoXuS/paper-datatable-api/commit/fa3607eee83e54e84b817e55b5e5d181cc38a1bd) to show you how you can add an element. ([See this file](https://github.com/RoXuS/paper-datatable-api/blob/master/demo/advanced-demo.html#L51)).

I hope it will help you.

**Comment by emilianosantucci** [comment]
> Hum, for me, the paper-datatable-api must be an independent component, maybe it is an error ?

Yes, it is my opinion too.
I only talked about David-Mulder's project because you "inspired" on it :).

> As you can see in the demo I have a toolbar to hide/show column, I have added paper-input in this commit to show you how you can add an element. (See this file).

I'm sorry, on demo have seen "only" hide/show column without other inputs on advanced usage.

To clarify my request, I try to explain it with an example.

As you can see on `paper-datatable`, we have the possibility to slice the toolbar declaration, using `toolbar-main` property on a _div_ element.
Furthermore we have possibility to separate _single select_ actions by _multiple select_ actions, respectively with `toolbar-select-single` and `toolbar-select`.

Especially in applications with a backoffice area, this is useful for create custom behaviours.

Currently I'm using `paper-datatable` on my Angular2 project but I had to bit change the original web component to run with it. Also, it have some bug with pagination behaviour as opposed to your component :).

In short, I really want to change my custom web component with your version but to reach this I need similar functionality of `paper-datatable` like custom toolbar, pagination events (ex. next, previous) and pagination buttons (first, last).

I realize that is not simple but I'm curious if my expectations merge with your "roadmap".

Thanks for all.

P.S.: Sorry for my english, I'm coding better than writing issues :D

**Comment by RoXuS** [comment]
Hi @emilianosantucci,

Finally do you use `paper-datatable-api` ?

Sorry I never answer to your questions....

About pagination events, we can add this but if you set an observer to the page property, you will have the event!
We can add `first` and `last` button too.
Do you want these features ?

About custom toolbar, `paper-datatable-api` doesn't have a toolbar, what do you want put in this toolbar ?

**Comment by emilianosantucci** [comment]
Hi @RoXuS,
thank you for your quick response :P :D.

Currently I've replaced paper-datatable with "cards" but is a temporary solution.

About your question, David create all-in-one solution for CRUD functionalities but his project seem to be abandoned :(.
In this use case, feature as toolbar, pagination, live editing, etc. take UX powerful. Look into toolbar, edit or delete multi elements in the list allow to handle "bulk" data change.

In my opinion inherit that features and events will grow your project's popularity and take it as *evolution* of a great (but dead) `paper-datatable` project.

I give you my best wish and waiting for it :)

**Comment by RoXuS** [comment]
Ok I see @emilianosantucci.

It is very simple to create a CRUD with my table.

An example with the update feature :  https://github.com/RoXuS/paper-datatable-api/blob/master/demo/edit-demo.html and the live demo on https://roxus.github.io/paper-datatable-api/components/paper-datatable-api/#paper-datatable-api.

Indeed, it is possible to add this feature directly in the `paper-datatable-api` but this element doesn't edit data like `paper-datatable` so I think is not mandatory...

What do you think about my example ?

**Comment by emilianosantucci** [comment]
I like it :D

I hope to migrate on paper-datatable-api asap and take my contribution on it.

Great work @RoXuS
